pkg/controller/directimagemigration/validation.go

```diff
@@ -0,0 +1,265 @@
+/*
+Copyright 2020 Red Hat Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package directimagemigration
+
+import (
+	"context"
+	"fmt"
+	"path"
+	"reflect"
+
+	liberr "github.com/konveyor/controller/pkg/error"
+	migapi "github.com/konveyor/mig-controller/pkg/apis/migration/v1alpha1"
+	migref "github.com/konveyor/mig-controller/pkg/reference"
+	kapi "k8s.io/api/core/v1"
+	k8serror "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// Types
+const (
+	InvalidSourceClusterRef               = "InvalidSourceClusterRef"
+	InvalidDestinationClusterRef          = "InvalidDestinationClusterRef"
+	InvalidSourceCluster                  = "InvalidSourceCluster"
+	InvalidDestinationCluster             = "InvalidDestinationCluster"
+	SourceClusterNotReady                 = "SourceClusterNotReady"
+	DestinationClusterNotReady            = "DestinationClusterNotReady"
+	MissingSourceClusterRegistryPath      = "MissingSourceClusterRegistryPath"
+	MissingDestinationClusterRegistryPath = "MissingDestinationClusterRegistryPath"
+	NsListEmpty                           = "NamespaceListEmpty"
+	NsNotFoundOnSourceCluster             = "NamespaceNotFoundOnSourceCluster"
+
+	Running   = "Running"
+	Failed    = "Failed"
+	Succeeded = "Succeeded"
+)
+
+// Reasons
+const (
+	NotFound    = "NotFound"
```

Comment:
Another general issue, sorry on picking this PR, these conts should probably be in the common-lib and imported into all the controllers, instead of being declared again and again